Revenant Forms

Armhulen opened the pull request:
:cl: Armhulen for code, Nilons for early on sprite placeholders, Onule for poltergeist final, and Anonmare for ideas
add: Revenants now take different forms based on how they died (hi, sims!!)
add: Brute damage: poltergeist, a violent red ghost with very physical interactions (throwing, shaking, maybe explosions?)
add: Burn damage: specter, a vibrant orange ghost with abilities that affect the atmosphere, affecting gas and temperature. Maybe stuff like freezing touch, fireburst, so on.
add: Toxin damage: wight, a sickly green ghost, would have abilities that impair or hurt without being noticed; covertly injecting some toxins, a contagion maybe.
add: Oxygen damage: phantom, a transparent blue ghost, it would cause unconsciousness/sleep at the worst times, then suck the oxygen out of the air.
add: Brain damage: wendigo, a crazed pink or grey ghost , would cause hallucinations with precise control over them, and could possess bodies (any damage the body would take would reflect on the wendigo, not the owner.
add: Clone damage: preta, a malformed ghost, it drives one person to insanity and makes them their thrall. The thrall must work with the preta to build a massive meat effigy. Then, the preta feasts and becomes a monster with powers and size relating to how much it's eaten. If the effigy is destroyed, the preta must go through but whole process but not before finding and eating the devout to death.
/:cl:

I wouldn't usually pr something this early, but I did anyways for two reasons:
1. This is a massive project, I'd like to have the maintainers happy with it before I get deep into it.
2. This is one of those projects where I would infinitely praise any idea guy who wants to give some suggestions for powers

WHY?????
revenants always play the same and I love the idea of ghostbuster searching on what ghost type you're fighting so you can play around it. for example, you get attacked by a brain ghost and don the mighty tin cap to fight it.

it's also a fun reference to the sims~

Commit:
poltergeist part 1

Commit:
sneek peek

Shadowflame909 commented:
>revenant upgrade. OH SNAP ARM I LOVE YOU

Armhulen commented:
I might bring back some spells that I already had done from Revenant++ too, that project was fun

Armhulen commented:
also whoops let me push commits

davethwave commented:
This seems interesting

Nabski89 commented:
Apparition, Phantom, Spectre, Wraith?

Oni, Rusalka (more a drowning ghost in a place with no water).

https://en.wikipedia.org/wiki/Mare_(folklore)#Slavic_countries is a fun one. There's actually a bunch of fun ones if you trawl through the slavic section.

Anonmare commented:
Will the ghost forms be random or pulled from the ghost's body? If so, if they never had a body what happens?

Suggested names: 
Poltergeist
Wight
Shade
Spectre
Phantom 
Phantasm

Buggy123 commented:
I like this but can it please be merged this time? I also liked revs++ and that was rejected.

fludd12 commented:
Totally taking a page from Whitewolf games, but:
Brute: `The Torn`
Burn: `The Forgotten`
Tox: `The Stricken`
Oxy: `The Silent`
Brain: `The Geist`
Clone: `The Prey`

Commit:
Poltergeist abilities working!

CameronWoof commented:
An important thing about Rev's current design is that their abilities aren't precise. They can't affect one person, but not the person standing next to them. 

Discriminatory or precision abilities will allow the poltergeist to team up with or against other antagonists, essentially playing kingmaker by pushing 1v1s or even small group scuffles in their favor. Don't know if that's a good thing or a bad thing, but it *is* 'invisible power' that can't reliably be planned for or worked against.

Armhulen commented:
Revenants can already use transmit to tell sec where antags are, and antags where sec/other antags/targets are. I'm not too worried. Besides, the powers given so far give no advantage to anyone teaming but i'll keep it in mind

davethwave commented:
Or there could be objectives such as take revenge on those who harmed you and no one else. Or "as you died protecting the station your soul has rose from the other side to be the publics protector

Anonmare commented:
Suggested abilities:

**Brute ghost abilities (that are not already listed in file changes):**
*Ethereal throw* - Basic ability that all brute ghosts start with. Spend a small amount of essence to target an unanchored object and throw it at the designated tile. Functions identically to TK and produces the same sparkles as it. Does not reveal.
*Calm* - Temporarily afflicts someone with the pacifism trauma, ideally used to make someone helpless against an attacker.
*Stigmata* - Causes open bleeding wounds to appear on the victim's body, very high bleeding but actual damage is surprisingly quite low. High cooldown as I feel direct damage should be avoided, ghosts should manipulate you or the environment to harm you.

**Burn ghost abilities:**
*Cold Spot* - Basic ability you start with. In a 3x3 radius, you cause the temperature to drop sharply by ~10 degrees. Does not reveal.
*Electrical malfunction* - Boobytrap a machine, the next time it is used it will shock someone. Boobytraps expire after 45 seconds. 
*Hot spot* - Causes the temperature to rise sharply in a 3x3 radius by ~10 degrees. Danger of causing flammables to spontaneously ignite.
*Spontaneous combustion* - You cause a person to burst into flames with the same amount of firestacks that stepping on lava in lavaland grants you. High cooldown.

**Toxin ghost abilities:**
*Fatigue* - Basic ability all toxin ghosts have. Afflict a target with a status effect that deals moderate stamina damage overtime that caps out at 60. Passes through your system in a few minutes. By itself it cannot kill but can make situations very worse. Does not reveal.
*Illness* - Force the target to vomit and gain a randomly generated illness of minor severity.
*Blight* - Afflict the target with a specific disease that causes toxin damage to build-up over time, in addition to causing narcolepsy. Can be cured by holy water but can never be immunised against.
*Pestilence* - Spread a severe rating disease to all living things within a 7x7 radius. Slight delay between the reveal and the spreading of the disease to give victims a chance at getting away. 

**Oxygen ghost abilities:**
*Deathgasp* - Basic ability all oxygen ghosts begin with. Makes a target briefly incapable of breathing, long enough to do ~60-70 oxy damage. Does not reveal. 
*Deprivation* - Deletes a set number of moles of oxygen from a tile in a 3x3 radius, literally pulls the air out of a room.
*Blackout* - Makes a target dizzy and confused, if they do not receive some form of stimulant - they will pass out for a moderate amount of time.
*Anemia* - Sets someone's blood level to be 60%. Very high cooldown.

**Brain ghost abilities:*
*Hallucination* - The default ability. Causes the target to suffer hallucinations of your choice for a few minutes. Can be used to make traiditonal hallucinations but can also cause the cult magic hallucination. Does not reveal. 
*Traumatise* - Temporarily afflict a person with a random severe trauma.
*Skinride* - Allows you to inhabit a living, nonbraindead body and pilot it as your own, you **CANNOT** make a person kill themselves or otherwise harm themselves directly. The possessed person will be told that they can RESIST to break free of your control.
Possessed people have an unnatural aura about them that manifest quite physically as a dark, smoke-like substance clinging to their body and speak in stilted sentences.
*Return* - Allows you to take control of a dead/catatonic body, a revenant if you will. Unlike possessing a living person, a Returned body knows it shouldn't exist and will slowly decay, faster if you take damage. A Returned body is very robust, being immune to stuns, not slowed down by pain, has no need to breathe or any use for blood. It can be delimbed like normal but it has a seperate health, quite high health bar of its own that decays over time. Once the meter hits - the body gets dusted. 
You can expend essence on a cooldown to keep the body going but it should be impossible to keep it permanently. Has the same aura as a possessed person but the examine text and medHUD will quite clearly tell you that this person is not Human.

**Clone ghost abilities:**
*Twisting flesh* - Makes you temporarily manifest a random, non-beneficial, non-benign mutation for a few minutes. Does not reveal.
*Malformed horror* - Revive a dead body as a malformed horror, a simple mob with a horrific sprite that attacks anything living it sees. Deadly but not very resilient, health about on par with a blob zombie. Chatters excitedly.
*Shapeless mass* - Creates a wall of flesh on the tile you're above. Flesh walls can be beaten down but cannot be deconstructed, whimpers in pain every now and then.
*Amalgamation* - Creates a slow but very tough simple mob on the tile you're above. Amalgamations are hostile to all living things and emit a minor amount of radioactivity in an aura around them. Resembles a puddle composed of human skin torn from several bodies. Constantly beg for death.

Armhulen commented:
IDEA

GUY

Armhulen commented:
THANK YOU

evsey9 commented:
IMO those abilities, especially spontaneous combust, makes revenant too powerful. Also we really should NOT have any target abilities that would allow you to be someone helping antags.

Ratvar commented:
"hey ghost when im near borgs or securetey empe them so i can bet the to death also where my trget johg green whisper plz and empe armorey" - revenant that already can help antags. Change combustion to some sort of plasma generation?

Profakos commented:
This gave me Ghost Master nostalgia flashbacks.

Bawhoppen commented:
These sound like they'd be really fun to sprite

Also all abilities really should reveal 
Otherwise I have no particular ideas about them

Bawhoppen commented:
Oh we could probably make them turn transparent slowly as their reveal duration shortens

Bawhoppen commented:
That's just a sprite thing that would require some code

KomradeSpectre commented:
Suggested name for the brain rev, Psychgeist.

Buggy123 commented:
TBH revenants helping traitors, etc is emergent gameplay. I've seen it happen before and its really not _that_ big of a deal, besides whats stopping 2 traitors, or a traitor and a ling, etc from teaming up?

Qustinnus commented:
add: Revenants now take different forms based on how they died (hi, sims!!)

bad moodies ghosts when?

Commit:
huge commit of changes

preta is mostly complete, moved most of the base spells to their respective ghost type, poltergeist testing sprites, rnadom revenant picking, thralls, effigies, OH MY!

Commit:
Thralls now get the escape objective when failing the effigy.

Commit:
Fixes for the fix god

kevinz000 commented:
Honestly side antags should stay weak

Wilchenx commented:
>side antags should stay weak
>blob
>xenos

Ratvar commented:
ninja 
abductors

arsserpentarium commented:
How does revenant spawn?

Commit:
Fixes part 2

Armhulen commented:
Like normal?

Commit:
Fixes part 3

Shadowflame909 commented:
>Honestly side antags should stay weak

Spiders, laughter demons, T-(REDACTED), Rogue Xenobiologists, Dumb Xenobiologists.

Commit:
Effigy requirements and leveling (spells and stats)

cebutris commented:
I think reventants from the more rare damage types (clone/brain) should be more powerful. And ghosts of suicides should be gimped somehow

Armhulen commented:
clone has a thrall and a meat abomination, brain has possess powers. revenants aren't picked from how you die

cebutris commented:
Oh. Right.

Armhulen commented:
clone and brain are still rarer though, if that counts

Commit:
Fixes part 4

Armhulen commented:
I've decided to work on energy parasites to really stabilize that feeling of wanting to die now that revenants take a more frontal role

Sentient Powersinks, WHY DIDNT I THINK OF THIS BEFORE

Commit:
fixes

Commit:
merge conflicts?

Commit:
Merge branch 'master' into patch-53

Commit:
fixes spacing, renames some poltergeist abilities and adds a bit of work on a new one

Commit:
new sprites for poltergeist thank you onule!

Armhulen commented:
travis: owned.

AnturK commented on code/modules/antagonists/revenant/revenant.dm:
```diff
@@ -60,18 +60,47 @@
-	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/revenant/blight(null))
-	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/revenant/malfunction(null))
 	random_revenant_name()
+	if(type=="/mob/living/simple_animal/revenant") //so not an admin spawned specific revenant, lets randomly pick one
```
No.

ShizCalev replied:
.....

Armhulen replied:
yeah yeah there's a better way to do it

but like why not this one line check

ShizCalev replied:
there shouldn't be any quotes for one, if someone changes the path and this is missed, then it'll silently fail since it's a string.

AnturK commented on code/modules/antagonists/revenant/revenant.dm:
```diff
@@ -60,18 +60,47 @@
+/mob/living/simple_animal/revenant/proc/pickghost()
+	switch(rand(1,15))
+		if(1 to 3) //brute
+			pickedghost = "Poltergeist"
```
Make this switch on ghost type (also use defines for these) then set that value in subtypes and only pick random if it is not set yet.

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+			to_chat(r, "<span class='revenwarning'>Get a thrall first!</span>")
+			return FALSE
+		to_chat(r, "<span class='revenboldnotice'>Fine place as any to prepare a feast!</span>")
+		//var/turf/placement = get_turf(user)
```
remove commented code

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+			return FALSE
+		to_chat(r, "<span class='revenboldnotice'>Fine place as any to prepare a feast!</span>")
+		//var/turf/placement = get_turf(user)
+		var/obj/structure/effigy/e = new(loc)
```
get_turf(user)

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+		else
+			to_chat(user, "<span class='revenboldnotice'>Here's our effigy, granted by our master. We need to start building a wonderful feast for it, and that requires...</span>")
+		for(var/obj/item/item in commonitem)
+			to_chat(user, "<span class='revenwarning'>one [item]!</span>")
```
Capitalize.

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -124,7 +126,65 @@
 			to_chat(ded, "[follow_rev] <span class='revenboldnotice'>[user] Revenant Transmit:</span> <span class='revennotice'>\"[msg]\" to</span> [follow_whispee] <span class='name'>[M]</span>")
 
+/obj/effect/proc_holder/spell/targeted/revenant
+	clothes_req = 0
```
FALSE

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -124,7 +126,65 @@
+	return TRUE
 
 /obj/effect/proc_holder/spell/aoe_turf/revenant
 	clothes_req = 0
```
FALSE

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+	var/maxthrow = 5
+	var/sparkle_path = /obj/effect/temp_visual/gravpush
+	var/anti_magic_check = TRUE
+	action_icon_state = "repulse"
```
put this above the the new vars.

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+	stun = 0
+
+///obj/effect/proc_holder/spell/targeted/revenant/punch/cast()
+//	if(attempt_cast(user))
```
remove ded code

Armhulen replied:
I'll finish it sheesh

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+	cast_amount = 40
+	var/shock_range = 2
+	var/shock_damage = 15
+	action_icon_state = "overload_lights"
```
above the rest

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/overload/proc/overload(turf/T, mob/user)
+	for(var/obj/machinery/light/L in T)
+		if(!L.on)
```
if(L.turned_off() || !(L.has_power() || L.has_emergency_power()))

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+		addtimer(CALLBACK(src, .proc/overload_shock, L, user), 20)
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/overload/proc/overload_shock(obj/machinery/light/L, mob/user)
+	if(!L.on) //wait, wait, don't shock me
```
if(L.turned_off() || !(L.has_power() || L.has_emergency_power()))

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+	locked = FALSE
+
+/obj/effect/proc_holder/spell/targeted/revenant/enthrall/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(attempt_cast(user))
```
early return

if(!istype(user))
return FALSE

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+
+/obj/effect/proc_holder/spell/targeted/revenant/enthrall/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(attempt_cast(user))
+		if(!isrevenant(user))
```
then remove this

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+	if(attempt_cast(user))
+		if(!isrevenant(user))
+			return FALSE
+		var/mob/living/simple_animal/revenant/r = user
```
user is already typecasted

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/effigy/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(attempt_cast(user))
+		if(!isrevenant(user))
```
early return

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+	if(attempt_cast(user))
+		if(!isrevenant(user))
+			return FALSE
+		var/mob/living/simple_animal/revenant/r = user
```
already typecasted

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+	completed = 1
+
+/datum/objective/thrall/effigy
+	completed = 0
```
FALSE

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,256 @@
+	return report.Join("<br>")
+
+/datum/objective/thrall
+	completed = 1
```
TRUE

Commit:
shiz fixes...

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+		addtimer(CALLBACK(src, .proc/overload_shock, L, user), 20)
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/overload/proc/overload_shock(obj/machinery/light/L, mob/user)
+	if(L.turned_off() || !(L.has_power() || L.has_emergency_power()))) //wait, wait, don't shock me
```
too many closing brackets at the end.

Armhulen replied:
lmao

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -379,3 +530,254 @@
+	if(attempt_cast(user))
+		if(!isrevenant(user))
+			return FALSE
+		var/mob/living/simple_animal/revenant/r = user
```
user is already typecasted as a revenant

ShizCalev commented on code/modules/antagonists/revenant/revenant_abilities.dm:
```diff
@@ -326,6 +345,138 @@
+	stun = 0
+
+///obj/effect/proc_holder/spell/targeted/revenant/punch/cast(list/targets, mob/user = usr)
+	//if(attempt_cast(user))
```
dead code?

Commit:
Merge branch 'patch-53' of https://github.com/Armhulen/tgstation into patch-53

XDTM commented:
>Anonmare for ideas

ree where's my credits

Armhulen commented:
The amount of sprites required is just colossal and I just don't have them

epicmemename commented:
:sob:

davethwave commented:
dang...
